pandas/_libs/tslibs/resolution.pyx

```diff
@@ -39,51 +38,42 @@ def get_resolution(const int64_t[:] stamps, tzinfo tz=None):
         Py_ssize_t i, n = len(stamps)
         npy_datetimestruct dts
         int reso = RESO_DAY, curr_reso
-        ndarray[int64_t] trans
-        int64_t[:] deltas
-        Py_ssize_t[:] pos
-        int64_t local_val, delta
-
-    if is_utc(tz) or tz is None:
-        for i in range(n):
-            if stamps[i] == NPY_NAT:
-                continue
-            dt64_to_dtstruct(stamps[i], &dts)
-            curr_reso = _reso_stamp(&dts)
-            if curr_reso < reso:
-                reso = curr_reso
-    elif is_tzlocal(tz):
+        int64_t local_val
+        TZConvertInfo info
+        ndarray[intp_t, ndim=1] pos2
+
+    info = get_tzconverter(tz, stamps)
+
+    if info.use_fixed:
+        assert info.delta != NPY_NAT
+    elif not info.use_utc and not info.use_tzlocal:
+        assert info.utcoffsets is not NULL
+        assert info.positions is not NULL
+        pos2 = np.array(<intp_t[:n]>info.positions, dtype=np.intp)
```

Comment:
If I had to guess this cast is not a safe cast (think we saw this on another PR) to convert a int pointer to a memory view

Reply:
This is just for debugging, not part of the original trouble

Reply:
Do you get the feeling that the issue is really with an overflow on `info.positions`? Like the assertion that `v1 < info.noffsets` is out of whack because v1 is much larger than it should be right?

If that's the case, it may be the cast down on line 258 `info.positions = <intp_t*>cnp.PyArray_DATA(pos)`

Reply:
Nvm I see what you are doing there...hmm still looking

FYI I don't think you need to use cnp.PyArray_DATA. You can still read from `.data` just writing to it is what doesn't work in newer versions of Cython / NumPy

Reply:
> I don't think you need to use cnp.PyArray_DATA

Huh, when I do just `self.positions = pos.data`, I get a compile-time error "Cannot convert python object to intp_t*".  when I declare `pos` as `ndarray[intp_t, ndim=1]` that error becomes `Cannot assign type 'char *' to 'intp_t *'`

Reply:
> it may be the cast down on line 258 

Why is that a cast down?

Reply:
> Huh, when I do just `self.positions = pos.data`, I get a compile-time error "Cannot convert python object to intp_t*". when I declare `pos` as `ndarray[intp_t, ndim=1]` that error becomes `Cannot assign type 'char *' to 'intp_t *'`

Yea you still need the cast. Whether using the C macro or `.data` NumPy still generically stores everything as a pointer to a char, so need to cast to the correct type in both cases

Reply:
> Why is that a cast down?

Its not I think my wording was just confusing. Remove the word "down" from the statement and the context is correct